### PR TITLE
update the entrypoint script and use NODE_EXTRA_CA_CERTS if provided

### PR DESCRIPTION
**Pull request by dovholuknf**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -341,10 +341,15 @@ ${thirdPartyHTML}
     selects.push(formselect);
     /** -------------------------------------------------------------------------------------------------- */
 
-    // Make sure we don't experience the dreaded UNABLE_TO_VERIFY_LEAF_SIGNATURE issue when we make REST calls to the Controller
-    https.globalAgent.options.ca = fs.readFileSync(
-        'node_modules/node_extra_ca_certs_mozilla_bundle/ca_bundle/ca_intermediate_root_bundle.pem'
-    );
+	// Check if NODE_EXTRA_CA_CERTS is set
+	const extraCaCertsPath = process.env.NODE_EXTRA_CA_CERTS;
```
Comment (rentallect) [inline index.js:345]: pls change to `env('NODE_EXTRA_CA_CERTS')`

**Review by rentallect — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -341,10 +341,15 @@ ${thirdPartyHTML}
...
+	if (extraCaCertsPath) {
+	  // Read the CA certificates from the file specified in the environment variable
+	  https.globalAgent.options.ca = fs.readFileSync(extraCaCertsPath);
+	  console.log(`Using CA certificates from env var: ${extraCaCertsPath}`);
+	} else {
+	  console.log('NODE_EXTRA_CA_CERTS is not set. Using default CA certificates.');
+	}
```
[the remark this answers is not in the document]
Comment (dovholuknf) [reply index.js:352]: it's set in the entrypoint script now. if `NODE_EXTRA_CA_CERTS` isn't supplied, it'll get defaulted to `node_modules/node_extra_ca_certs_mozilla_bundle/ca_bundle/ca_intermediate_root_bundle.pem`

File: index.js (hunk 1 above)
Comment (dovholuknf) [reply index.js]: all set

**Comment by dovholuknf** [comment]
> You can add the default value for the `NODE_EXTRA_CA_CERTS` var in `env.js` near line `52`. Doing so is tidy'er than hacking on the docker entrypoint file, and will also be applicable to non-docker bootstrapper start up's.

Ok, I'll change it... Just curious, are you sure it's necessary still? I thought it was only used because the trust store in the alpine image didn't contain a fully up-to-date set of certs?

File: index.js [hunk 3]
```diff
@@ -44,6 +44,7 @@ var getAccessToken = require('./lib/oidc/utils').getAccessToken;
 var ZITI_CONSTANTS = require('./lib/edge/constants');
 const Mustache = require('mustache');
 const he = require('he');
+var nconf       = require('nconf');
```
Comment (rentallect) [inline index.js:47]: this is unnecessary.

File: index.js [hunk 4]
```diff
@@ -342,9 +343,9 @@ ${thirdPartyHTML}
     /** -------------------------------------------------------------------------------------------------- */
 
     // Make sure we don't experience the dreaded UNABLE_TO_VERIFY_LEAF_SIGNATURE issue when we make REST calls to the Controller
-    https.globalAgent.options.ca = fs.readFileSync(
-        'node_modules/node_extra_ca_certs_mozilla_bundle/ca_bundle/ca_intermediate_root_bundle.pem'
-    );
+    const extraCaCertsPath = nconf.get('NODE_EXTRA_CA_CERTS');
```
Comment (rentallect) [inline index.js:346]: this should be `env('NODE_EXTRA_CA_CERTS')`

**Review by rentallect — CHANGES_REQUESTED** [review]

**Review by rentallect — APPROVED** [review]

**Pull request merged**